platform_plugin_ontask/api/utils.py

```diff
@@ -0,0 +1,62 @@
+"""Utility functions for the OnTask plugin API."""
+
+from __future__ import annotations
+
+from importlib import import_module
+
+from django.conf import settings
+from rest_framework.response import Response
+
+from platform_plugin_ontask.datasummary.backends.base import DataSummary
+from platform_plugin_ontask.datasummary.backends.completion import CompletionDataSummary
+
+
+def api_field_errors(field_errors: dict, status_code: int) -> Response:
+    """
+    Build a response with field errors.
+
+    Args:
+        field_errors (dict): Errors to return.
+        status_code (int): Status code to return.
+
+    Returns:
+        Response: Response with field errors.
+    """
+    return Response(data={"field_errors": field_errors}, status=status_code)
+
+
+def api_error(error: str, status_code: int) -> Response:
+    """
+    Build a response with an error.
+
+    Args:
+        error (str): Error to return.
+        status_code (int): Status code to return.
+
+    Returns:
+        Response: Response with an error.
+    """
+    return Response(data={"error": error}, status=status_code)
```

Comment:
Is having two different API errors necessary?

Reply:
`api_field_errors` returns an error from a specific field of the request, e.g. in the `course_id`, while `api_error` returns a more general error unrelated to any field. However, we could leave one that accepts both types of errors.